src/mongoc/mongoc-topology-scanner.c

```diff
@@ -398,13 +399,18 @@ mongoc_topology_scanner_start (mongoc_topology_scanner_t *ts,
 
    DL_FOREACH_SAFE (ts->nodes, node, tmp)
    {
-      if (mongoc_topology_scanner_node_setup (node)) {
+      /* FIXME: Gotta catch all the pokemons, not overwrite them */
+      bson_error_t error;
```

Comment:
I think we should fill the remainder of the bson_error_t's 504-byte error message with a list of _mongoc_server_description_t.error strings, or provide a way to retrieve those errors from a topology. If there are no errors but we still can't select a server, just print a list of server types: "Can't select a server for read preference "primary", with servers of type SECONDARY, SECONDARY, ARBITER." If there are *no* servers of known type, then "All servers are down."